openlibrary/plugins/openlibrary/js/autocomplete.js

```diff
@@ -16,59 +16,118 @@ export default function($) {
+                '<strong>$1</strong>'
+            );
+        }
 
-                $(`#${this.id}-key`).val(item.key);
-                $this = $(this);
+        $.widget('custom.autocompleteHTML', $.ui.autocomplete, {
+            _renderMenu($ul, items) {
+                $ul.addClass('ac_results').attr('id', this.ulRef);
+                items.forEach((item, i) => {
+                    $('<li>')
+                        .addClass(i % 2 ? 'ac-even' : 'ac-odd')
+                        .attr('data-value', item.value)
+                        .data('ui-autocomplete-item', item)
+                        .attr('aria-label', item.value)
+                        .html(item.label)
+                        .appendTo($ul);
+                });
+                // store list so we can add ac_over hover effect in `focus` event
+                $(_this).data('list', $ul);
+            }
+        });
 
-                //adding class directly is not working when tab is pressed. setTimeout seems to be working!
-                setTimeout(function() {
-                    $this.addClass('accept');
-                }, 0);
-            })
-            .nomatch(function(){
-                $(`#${this.id}-key`).val('');
-                $(this).addClass('reject');
-            })
-            .keypress(function() {
+        const options = $.extend(default_ac_opts, ac_opts, {
+            source: function (q, response) {
+                const term = q.term;
+                return $.ajax({
+                    url: ol_ac_opts.endpoint,
+                    data: {
+                        q: term,
+                        limit: options.max,
+                        timestamp: new Date()
+                    }
+                }).then((results) => {
+                    const modifiedResults = results.map((r) => {
+                        return {
+                            key: r.key,
+                            label: highlight(options.formatItem(r), term),
+                            value: r.name
+                        };
+                    });
+
+                    // When no results if callback is defined, append a create new entry
+                    if (
+                        ol_ac_opts.addnew === true ||
+                        (ol_ac_opts.addnew && ol_ac_opts.addnew(term))
+                    ) {
+                        response(
+                            modifiedResults.concat([
+                                {
+                                    label: options.formatItem({
+                                        name: term,
+                                        key: '__new__',
+                                        result: term,
+                                        value: term
+                                    }),
+                                    value: term
+                                }
```

Comment:
```suggestion
                                {
                                    key: '__new__',
                                    label: options.formatItem({
                                        name: term,
                                        key: '__new__',
                                        result: term,
                                        value: term
                                    }),
                                    value: term
                                }
```

Reply:
Right. Yeh the key doesn't need to be there. i've refactored this code a little.